scripts/skills/perks/perk_rf_line_breaker.nut

```diff
@@ -15,24 +15,20 @@ this.perk_rf_line_breaker <- ::inherit("scripts/skills/skill", {
 	function onAdded()
 	{
 		this.getContainer().add(::new("scripts/skills/actives/rf_line_breaker_skill"));
+		local offhand = this.getContainer().getActor().getOffhandItem();
+		if (offhand != null) this.onEquip(offhand);
 	}
 
 	function onRemoved()
 	{
 		this.getContainer().removeByID("actives.rf_line_breaker");
 	}
 
-	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
+	function onEquip( _item )
 	{
-		if (_skill.getID() == "actives.knock_back" && _targetEntity.isAlive() && !_targetEntity.isDying())
+		if (::MSU.isKindOf(_item, "shield"))
```

Comment:
how does `isKindOf` work exactly?
Like what if there is a shield, that does not have `shield` in its classname?
And what about items, which do have `shield` in their name, but are not a shield? E.g. "Shieldbreaker"

Reply:
It means that `_item` must be an instance of the BB class `shield` or a descendant of it.